source/grid/manifold_lib.cc

```diff
@@ -478,16 +504,16 @@ get_new_point (const ArrayView<const Point<spacedim>> &vertices,
 
                   gradlocal[0] = cosphi;
                   gradlocal[1] = sinphi;
-                  gradient += (weights[i]*theta)*gradlocal;
+                  gradient += (merged_weights[i]*theta)*gradlocal;
 
                   const double sinphiSq = sinphi*sinphi;
                   const double cosphiSq = cosphi*cosphi;
-                  const double tt = weights[i]*(theta*sinthetaInv)*costheta;
-                  const double offdiag = cosphi*sinphi*(weights[i]-tt);
-                  Hessian[0][0] += weights[i]*cosphiSq+tt*sinphiSq;
+                  const double tt = (theta*sinthetaInv)*costheta;
+                  const double offdiag = cosphi*sinphi*merged_weights[i]*(1.0-tt);
+                  Hessian[0][0] += merged_weights[i]*(cosphiSq+tt*sinphiSq);
                   Hessian[0][1] += offdiag;
                   Hessian[1][0] += offdiag;
-                  Hessian[1][1] += weights[i]*sinphiSq+tt*cosphiSq;
+                  Hessian[1][1] += merged_weights[i]*(sinphiSq+tt*cosphiSq);
```

Comment:
Thanks for the micro-optimizations!

Reply:
Well, once you have looked enough at code that nicely transforms to FMAs you try to set that pattern everywhere... :smiley: